File: module-domain/src/main/java/com/example/entity/reservation/Seat.java
package com.example.entity.reservation;

import com.example.entity.BaseEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import org.springframework.lang.Nullable;

import java.time.LocalDateTime;

@Entity
public class Seat extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long theaterId;

    @Nullable
    private Long userId;

    private char seatRow;

    private Long seatCol;

    private Boolean isReservation;

    private LocalDateTime reservationTime;
}

File: module-domain/src/main/kotlin/com/example/redis/theater/Seat.kt
package com.example.redis.theater

class Seat(
    val seatId: Long,
    val seatRow: String,
    val seatCol: String,
) {
    fun isSameRow(otherSeat: Seat): Boolean {
        return seatRow == otherSeat.seatRow
    }
}